csharp/ql/src/semmle/code/csharp/security/dataflow/flowsinks/Html.qll

```diff
@@ -21,21 +22,23 @@ private import semmle.code.asp.AspNet
  */
 abstract class HtmlSink extends DataFlow::ExprNode, RemoteFlowSink { }
 
+private class ExternalHtmlSink extends HtmlSink {
+  ExternalHtmlSink() { sinkNode(this, "html") }
+}
+
 /**
  * An expression that is used as an argument to an HTML sink method on
  * `HttpResponse`.
  */
-class HttpResponseSink extends HtmlSink {
-  HttpResponseSink() {
-    exists(Method m, SystemWebHttpResponseClass responseClass |
-      m = responseClass.getAWriteMethod() or
-      m = responseClass.getAWriteFileMethod() or
-      m = responseClass.getATransmitFileMethod() or
-      m = responseClass.getABinaryWriteMethod()
-    |
-      // Calls to these methods, or overrides of them
-      this.getExpr() = m.getAnOverrider*().getParameter(0).getAnAssignedArgument()
```

Comment:
We can't express `getAnOverrider*` in CSV, but `HttpResponse` is `sealed` anyway.